plugins/parsers/csv/parser_test.go

```diff
@@ -322,3 +323,30 @@ func TestParseStream(t *testing.T) {
 			DefaultTime(),
 		), metric)
 }
+
+func TestTimestampUnixFloatPrecision(t *testing.T) {
+	p := Parser{
+		MetricName:      "csv",
+		ColumnNames:     []string{"time", "value"},
+		TimestampColumn: "time",
+		TimestampFormat: "unix",
+		TimeFunc:        DefaultTime,
+	}
+	data := `1551129661.95456123352050781250,42`
+
+	expected := []telegraf.Metric{
+		testutil.MustMetric(
+			"csv",
+			map[string]string{},
+			map[string]interface{}{
+				"value": 42,
+				"time":  1551129661.954561233,
```

Comment:
I think this is a bug, we should be removing the time field when it is used as a timestamp column.  I will fix this in a separate pr for 1.11.